packages/core/core/src/PackagerRunner.js

```diff
@@ -150,7 +150,10 @@ function generateDepToBundlePath(
     let entryBundle = entryBundleNode.value;
     depToBundlePath.set(
       dep.id,
-      urlRelative(nullthrows(bundle.name), nullthrows(entryBundle.name))
+      urlJoin(
```

Comment:
one thing we could try is using a url relative to the current bundle only in the case that `publicUrl` wasn't provided by the user, and if it was, use a url relative to that. What do you think?